### PR TITLE
feat(LTTStore): add presence

### PR DESCRIPTION
**Pull request by AtlasWuff**
## Description 

Newly created presence for https://lttstore.com

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

image
image
image
image

</details>

### CODE REVIEW

File: websites/L/LTTStore.com/metadata.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		"id": "484343723426054150",
+		"name": "atlaswuff"
+	},
+	"service": "LTTStore.com",
+	"description": {
+		"en": "The official merch store of Linus Tech Tips"
+	},
+	"url": "lttstore.com",
+	"version": "1.0.0",
+	"logo": "https://www.lttstore.com/cdn/shop/files/LTT_Logo.png",
```
Comment (DarkVillager) [inline websites/L/LTTStore.com/metadata.json:13]: Logo must be 512x512

File: websites/L/LTTStore.com/presence.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
+const presence = new Presence({
+	clientId: "1258619078201839646",
+});
+
+const enum Assets { // Other default assets can be found at index.d.ts
+	Logo = "https://www.lttstore.com/cdn/shop/files/LTT_Logo.png",
```
Comment (DarkVillager) [inline websites/L/LTTStore.com/presence.ts:6]: Ditto

File: websites/L/LTTStore.com/presence.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: Assets.Logo,
+		buttons: [
+			{
+				label: "Visit LTTStore",
+				url: "https://www.lttstore.com/",
+			},
+		],
```
Comment (DarkVillager) [inline websites/L/LTTStore.com/presence.ts:17]: Can't advertise like this using buttons;

File: websites/L/LTTStore.com/presence.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	};
+
+	if (
+		document.location.hostname === "www.lttstore.com" ||
+		document.location.hostname === "lttstore.com"
+	) {
```
Comment (DarkVillager) [inline websites/L/LTTStore.com/presence.ts:23]: No need for this, due to the url in metadata no other hostnames will use the presence.

File: websites/L/LTTStore.com/presence.ts [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+					for (let i = 0; i < firstImgSrcSplit.length; i++) {
+						if (firstImgSrcSplit[i] !== "/") {
+							firstImgSrcSplit.splice(0, i);
+							break;
+						}
+					}
+				}
+			}
+
+			presenceData.largeImageKey = `https://${firstImgSrcSplit.join("")}`;
```
Comment (DarkVillager) [inline websites/L/LTTStore.com/presence.ts:47]:
```suggestion
			presenceData.largeImageKey = document.querySelector<HTLMImageElement>('[property="og:image"]')?.src ?? Assets.Logo
```

File: websites/L/LTTStore.com/presence.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+				presenceData.state = "Collections";
+			}
+		} else if (document.location.pathname.includes("/blogs/")) {
+			presenceData.details = "Reading";
+			presenceData.state = document.querySelector(
+				"h1.article-template__title"
+			)?.textContent;
+		} else if (document.location.pathname.includes("/cart")) {
+			presenceData.details = "Viewing Cart";
+			presenceData.state = "";
```
Comment (DarkVillager) [inline websites/L/LTTStore.com/presence.ts:75]: We prefer deleting over setting to ""
```suggestion
			delete presenceData.state
```

File: websites/L/LTTStore.com/presence.ts [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+		} else {
+			presenceData.details = "Browsing";
+			presenceData.state = "";
```
Comment (DarkVillager) [inline websites/L/LTTStore.com/presence.ts:78]: We prefer deleting over setting to ""
```suggestion
			delete presenceData.state 
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/L/LTTStore.com/metadata.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.10",
+	"author": {
+		"id": "484343723426054150",
+		"name": "atlaswuff"
+	},
+	"service": "LTTStore.com",
```
Comment (theusaf) [inline websites/L/LTTStore.com/metadata.json:7]: Based on the website's information, the service should just be "LTTStore"

File: websites/L/LTTStore.com/presence.ts [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+
+const enum Assets {
+	Logo = "https://upload.wikimedia.org/wikipedia/commons/thumb/7/77/2018_Linus_Tech_Tips_logo.svg/512px-2018_Linus_Tech_Tips_logo.svg.png",
+}
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: Assets.Logo,
+	};
+	if (document.location.pathname === "/" || document.location.pathname === "") {
```
Comment (theusaf) [inline websites/L/LTTStore.com/presence.ts:13]: The second condition is impossible. 
```suggestion
	if (document.location.pathname === "/") {
```

File: websites/L/LTTStore.com/presence.ts [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+	} else if (document.location.pathname.includes("/products/")) {
+		presenceData.largeImageKey =
+			document.querySelector<HTMLMetaElement>('[property="og:image"]')
+				?.content ?? Assets.Logo;
+
+		presenceData.smallImageKey = Assets.Logo;
+		presenceData.smallImageText = "LTTStore.com";
+		presenceData.details = "Viewing Product";
+		presenceData.state = (
+			document.querySelector("div.product__title h1") as HTMLElement
```
Comment (theusaf) [inline websites/L/LTTStore.com/presence.ts:25]: Use generics rather than casting. In this case, a specific type might not be needed.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/L/LTTStore.com/presence.ts (hunk 10 above)
Comment (AtlasWuff) [reply websites/L/LTTStore.com/presence.ts]: Im not entirely sure as to what you mean. I put in the commit what I believe is what you mean but I'll leave it up to review

**Review by theusaf — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**